lib/src/models/comment_impl.dart

```diff
@@ -150,10 +149,56 @@ class MoreComments extends RedditBase with RedditBaseInitializedMixin {
       _comments = await reddit.post(apiPath['morechildren'], data);
 
       if (update) {
-        _comments.forEach((c) => c._submission = _submission);
+        _comments.forEach((c) {
+          c._submission = _submission;
+        });
       }
     }
-    return _comments;
+    return _fillCommentsForests(_comments);
+  }
+
+  List<dynamic> _fillCommentsForests(List<dynamic> fullList) {
+    final first = fullList.first;
+    if (fullList.length > 0 && first is Comment) {
+      currentIndex = 0;
+      return _fillCommentsForestsRecursively(fullList, first.depth);
+    } else {
+      return fullList;
```

Comment:
Can you give an example of which case would result in us just returning `fullList`?

Reply:
As mentioned above  
[MoreComments]  

But I guess this should also catch  
[Comment]